### ISSUE TITLE
Dataset is not opening

### ISSUE THREAD

**Issue by kanika02** [FudanVI/FudanOCR#13]
I was trying to open the .mdb dataset files, but I could not access and read them. I have MS access, and I tried other options too. Could you help me out?

**Comment by JingyeChen** [comment]
Hello, the .lmdb files can not be directly accessed.
To visualize the images in .mdb，you could refer to the Class lmdbDataset_real at
scene-text-telescope/dataset/dataset.py

If you have other questions, feel free to leave an issue :D

**Comment by kanika02** [comment]
I am not able to visualize the images. Can you give more information of class lmdbDataset_real at dataset.py .

**Comment by JingyeChen** [comment]
import random
import torch
from torch.utils.data import Dataset
from torch.utils.data import sampler
import torchvision.transforms as transforms
import lmdb
import six
import sys
import bisect
import warnings
from PIL import Image
import numpy as np
import string

def buf2PIL(txn, key, type='RGB'):
    imgbuf = txn.get(key)
    buf = six.BytesIO()
    buf.write(imgbuf)
    buf.seek(0)
    im = Image.open(buf).convert(type)
    return im

class lmdbDataset(Dataset):
    def __init__(self, root=None, voc_type='upper', max_len=100, test=False):
        super(lmdbDataset, self).__init__()
        self.env = lmdb.open(
            root,
            max_readers=1,
            readonly=True,
            lock=False,
            readahead=False,
            meminit=False)

        if not self.env:
            print('cannot creat lmdb from %s' % (root))
            sys.exit(0)

        with self.env.begin(write=False) as txn:
            nSamples = int(txn.get(b'num-samples'))
            self.nSamples = nSamples
        self.voc_type = voc_type
        self.max_len = max_len
        self.test = test

    def __len__(self):
        return self.nSamples

    def __getitem__(self, index):
        assert index <= len(self), 'index range error'
        index += 1
        txn = self.env.begin(write=False)
        label_key = b'label-%09d' % index
        word = str(txn.get(label_key).decode())
        img_HR_key = b'image_hr-%09d' % index  # 128*32
        img_lr_key = b'image_lr-%09d' % index  # 64*16
        try:
            img_HR = buf2PIL(txn, img_HR_key, 'RGB')
            img_lr = buf2PIL(txn, img_lr_key, 'RGB')
        except IOError or len(word) > self.max_len:
            return self[index + 1]
        # label_str = str_filt(word, self.voc_type)
        label_str = word
        return img_HR, img_lr, label_str

# Visulize the dataset
def check_dataset(root, index=None):
    dataset = lmdbDataset(root)
    length = len(dataset)
    if index is None:
        for i in range(length):
            image_hr, image_lr, label = dataset[i]

            image_hr = image_hr.resize((128,32))
            image_lr = image_lr.resize((128,32))

            print('****')
            display(image_hr)
            display(image_lr)
            print(label)
    else:
        image_hr, image_lr, label = dataset[index]
        display(image_hr)

if __name__ == '__main__':
    check_dataset('./FudanOCR/scene-text-telescope/dataset/mydata/test/easy',467)

**Comment by kanika02** [comment]
Thank you so much, it was really helpful.

**Comment by kanika02** [comment]
Where and how can I see the generated images(result)  and where are the paths of the two images for which we are calculating psnr and ssim?

**Comment by kanika02** [comment]
![image](https://user-images.githubusercontent.com/35134378/134329038-dfa1178c-e2a7-497d-975e-8219802d61db.png)


getting error in the demo comand

**Comment by kanika02** [comment]
![image](https://user-images.githubusercontent.com/35134378/134333425-fc01b4b3-7e97-4c2c-a774-3ba578891edf.png)

Demo command is showing error ? How to resolve this
